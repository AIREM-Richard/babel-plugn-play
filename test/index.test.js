import pluginTester from 'babel-plugin-tester';
import HoistCommonString from '../src/index';

pluginTester({
  // 插件
  plugin: HoistCommonString,
  // 插件名，可选
  pluginName: 'hoist-common-string',
  // 插件参数，可选
  pluginOptions: {
    minCount: 2,
  },
  tests: {
    'using jest snapshots': {
      // 输入
      code: `const s1 = "foo";
      const s2 = "foo";

      const s3 = "bar";

      function f1() {
        const s4 = "baz";
        if (true) {
          const s5 = "baz";
        }
      }`,
      // 使用快照测试
      snapshot: true,
    },
  },
});