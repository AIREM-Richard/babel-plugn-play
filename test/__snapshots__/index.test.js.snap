// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hoist-common-string 1. using jest snapshots: 1. using jest snapshots 1`] = `

const s1 = "foo";
      const s2 = "foo";

      const s3 = "bar";

      function f1() {
        const s4 = "baz";
        if (true) {
          const s5 = "baz";
        }
      }

      ↓ ↓ ↓ ↓ ↓ ↓

const s1 = "foo";
const s2 = "foo";
const s3 = "bar";
function f1() {
  const s4 = "baz";
  if (true) {
    const s5 = "baz";
  }
}

`;
